instrumentation/lettuce/lettuce-5.1/library/src/main/java/io/lettuce/core/protocol/OtelCommandArgsUtil.java

```diff
@@ -22,19 +23,27 @@ public final class OtelCommandArgsUtil {
    */
   public static List<String> getCommandArgs(CommandArgs<?, ?> commandArgs) {
     List<String> result = new ArrayList<>();
+    StringCodec stringCodec = new StringCodec();
+
     for (SingularArgument argument : commandArgs.singularArguments) {
-      String value = argument.toString();
-      if (argument instanceof KeyArgument && value.startsWith("key<") && value.endsWith(">")) {
-        value = value.substring("key<".length(), value.length() - 1);
-      } else if (argument instanceof ValueArgument
-          && value.startsWith("value<")
-          && value.endsWith(">")) {
-        value = value.substring("value<".length(), value.length() - 1);
-      }
+      String value = getArgValue(stringCodec, argument);
       result.add(value);
     }
     return result;
   }
 
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private static String getArgValue(StringCodec stringCodec, SingularArgument argument) {
+    if (argument instanceof KeyArgument) {
+      KeyArgument keyArg = (KeyArgument) argument;
+      return stringCodec.decodeValue(keyArg.codec.encodeValue(keyArg.key));
+    }
+    if (argument instanceof ValueArgument) {
+      ValueArgument valueArg = (ValueArgument) argument;
+      return stringCodec.decodeValue(valueArg.codec.encodeValue(valueArg.val));
```

Comment:
Taken from the source of [`ValueArgument.toString()`](https://github.com/lettuce-io/lettuce-core/blob/a5804b86c54490aa0ca1d234c168bccbfff9f2c1/src/main/java/io/lettuce/core/protocol/CommandArgs.java#L747). One note: the codec is being reused here to avoid allocations.

Reply:
Yeah, agreed that this feels a tad more fragile. Hopefully it's a fair trade off for performance tho.